### ISSUE TITLE
Frame rate drops to many seconds per frame (intermittant)

### ISSUE THREAD

**Issue by Eribotes** [vegastrike/Vega-Strike-Engine-Source#760]
Occasionally, after jumping to a new system, the frame rate quickly slows from ~50-70+ fps to needing many seconds to update the screen and respond to key presses.  If you Esc & q before the frame rate drops too much you're able to exit the game but if you don't you then need to kill the game from another login.  The same problem seems to happen more often after taking off from a planet or station, and especially when you've just loaded a saved game that didn't exhibit any problems when you saved it.  In such cases I've just had to abandon that save and go back to an earlier one.

On a very few occasions, after jumping to a new system the frame rate has dropped from its normal ~50-70 fps but then stabilized at ~6-10 fps, enabling me to jump to another system, after which the frame rate rises back up to its normal level and everything is fine again.

The sound system seems to be running through the slow-downs and I can hear the messages from other ships when they destroy something but even then the message sometimes seem to be cut short.

Can't see anything obvious in the log but there are many repetitions of "The file is a directory !"

**Comment by Samweis2111** [comment]
Same here. The issue appears to be independent of the display and shader settings.
I had the impression that it occurs when there are loads of other ships in the same system, even if they were distant and hardly visible.

**Comment by Eribotes** [comment]
From the way that the frame rate slows down I initially suspected a runaway spawning process too but if it's spawning too many non-player ships then they're all too far away to be visible on the HUD scanner, which only shows a 'normal' number of ships.  I've certainly seen a lot more ships visible on the scanner without any associated problems, and with only a moderate drop in frame rate.

After letting the frame rate drop to > 1 second, but while I still thought I would be able to Esc & q (30 Secs),  I didn't see any sign of greatly increased memory usage either.  This just might mean that the spawned objects are just relatively small though.

**Comment by BenjamenMeyer** [comment]
@Samweis2111 @Eribotes what version are you using? Master? Or the release?
@royfalk could this be due to some of the refactors with the asset storage (assuming they're using master)?

**Comment by Samweis2111** [comment]
@BenjamenMeyer No, I did not compile current master. I just used the 0.8.0 release package for Debian bullseye.

**Comment by BenjamenMeyer** [comment]
> @BenjamenMeyer No, I did not compile current master. I just used the 0.8.0 release package for Debian bullseye.

Okay. We've had a lot of refactoring in master for 0.9.x.
If you can, would you please check to see if it still happens on master?
If not, any information on how to reproduce (hardware config, etc) would be greatly appreciated.

**Comment by Samweis2111** [comment]
@BenjamenMeyer: I am sorry - I am traveling right now and will not be back before June. All I can say from my memory is that the problem occurred on an ASUS Prime B550 Plus with Ryzen 3600, Radeon R5 230 graphics card, 16GB RAM. I know that's not really helpful, but if I installed VS on the laptop I have with me it would not surprise me if the graphics gets bogged down.
Hopefully @Eribotes can help out.

**Comment by fernando0010** [comment]
Can confirm this still happens (not always) in VS 0.9.1, Ubuntu 22.04.5 LTS.  After jumping to a new system the frame rate drops to the ~10fps range or even freezes. Not always but frequently.

**Comment by royfalk** [comment]
~I seem to recall seeing this issue even in the "original", pre current dev-team efforts. But it was a long time ago.~
~I suspect the issue is the reading of texture files from disk.~
~This issue isn't unique to Vegastrike. Other games prepare textures during game play and drop frames/freeze.~

~Unfortunately, this is currently not in my pipeline. The primary issue is that I am not familiar with the 3d graphics portion of the game and don't have the bandwidth to learn.~

This seems to be something else from what I thought. I assumed FPS would go up after a few seconds as I have observed when playing. @evertvorster , @kheckwrecker , have you experienced this?

**Comment by evertvorster** [comment]
I don't really see slowdowns, but then again I have not play tested extensively recently, and my video card is rather high end.

**Comment by kheckwrecker** [comment]
The only time I see a significant drop in framerate is in the middle of a jump. Once the jump is finished, the framerate jumps right back up to normal. Otherwise, I'm not experiencing any  issues.

________________________________

**Comment by stephengtuggy** [comment]
If you set your sim atom interval to a shorter interval than your hardware can handle, this can happen, and become a runaway problem that gets worse instead of better. The frame rate will drop, and then drop again, and then drop some more, until it's very close to zero. I have seen this happen many times, especially when running in a virtual machine.

**Comment by stephengtuggy** [comment]
Unfortunately, we probably won't have a fix for this anytime soon.

**Comment by stephengtuggy** [comment]
This problem is much rarer on 0.11.x, from what I have seen -- and possibly on 0.10.x as well. Both of these versions seem to have more efficient code that doesn't trip over itself nearly as much.

On 0.11.x/the master branch, this improvement is especially noticeable, with the conversion to SDL3 speeding things up considerably.

**Comment by stephengtuggy** [comment]
When I play 0.10.x now, I get 60 frames per second fairly consistently. When I run the code that's on master, I get more like 240-480 frames per second!
